### ISSUE TITLE
Vectorize calculation of axial currents

### ISSUE THREAD

**Issue by espenhgn** [LFPy/LFPy#148]
The calculation of axial currents (Algorithm 1 in https://doi.org/10.3389/fninf.2018.00092) is linear, that is, the `I = A*V_1 + B*V_2 + C*...` where the coefficients `A, B, C, ...` are computed from the corresponding compartment axial resistivity. The calculation of axial currents from membrane voltages can be reformulated as **I** = **MV** where **V** is a matrix with all voltages, **M** a sparse matrix with the corresponding coefficients that can be computed once. The calculation of axial currents can then be computed very fast without explicit for loops. 

@solveignaess, does this sound feasible?

**Comment by solveignaess** [comment]
Yes, I think this sounds feasible. However, finding the axial resistance is not so straightforward (needs to be set specifically for first or last segment in a section, and for segments attached to more than one segment), and I think some looping might be necessary in order to do this. But I don't think we need to loop through all segments of each section, so it seems like we can save time here!

**Comment by espenhgn** [comment]
Constructing the sparse matrix **M** will indeed involve nested for-loops I agree, but it only needs to be computed once, and wouldn't be slow relative to the time it takes to solve for membrane voltages and transmembrane currents in a typical simulation.

**Comment by solveignaess** [comment]
Yes, of course, now I see! We only need one M independent of the number of timesteps.=) I really think we should do this.
